### ISSUE TITLE
NEW RULE: no-unused-new-array

### ISSUE THREAD

**Issue by jpike88** [eslint/eslint#14739]
**Please describe what the rule should do:**
This rule should ensure that fresh arrays are actually used somewhere instead of just mutated.

**What new ECMAScript feature does this rule relate to?**

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)

**Provide 2-3 code examples that this rule will warn about:**

```js

// THIS SHOULD NOT BE ALLOWED. THE ARRAY ISNT BEING RETURNED.
function someFunction(){
const myArray = [];
myArray.push(4);
}

// THIS SHOULD BE ALLOWED. THE ARRAY MAY BE ALREADY ASSIGNED TO SOMETHING ELSE.
function someOtherFunction(){
const myArray = someArrayfromElsewhere;
myArray.push(4);
}




```

**Why should this rule be included in ESLint (instead of a plugin)?**
It's an easy mistake to have a new array that you're populating with the results of another using .push etc, and forgot to actually use that array later in the code.

**Are you willing to submit a pull request to implement this rule?**
Maybe

**Comment by mdjermanovic** [comment]
Hi @jpike88, thanks for the issue!

Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept new rules if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify. You can always make a custom rule and publish a plugin.

**Comment by jpike88** [comment]
Any tips on how to achieve this with existing rules?

**Comment by mdjermanovic** [comment]
I don't think this can be achieved with existing rules, it would have to be a new rule.
